### ISSUE TITLE
Add support for specifying "clear" on all other blocks

### ISSUE THREAD

**Issue by raducoravu** [oxygenxml/dita-image-float#2]
Feedback from @ToshihikoMakita:

Introducing "float" needs more indispensable attribute "clear" that defines the block-level element to be moved down from the floated graphic.

The important point is "clear" is needed for any block elements.
So I've specialized "clear" attribute from "base". attribute.

AntennaHouse/ah-dita
https://github.com/AntennaHouse/ah-dita/blob/master/com.antennahouse.dita.dita13.doctypes/doctypes/dtd/base/dtd/clearAttDomain.ent

As long as for PDF output, it works fine by implementing "clear" attribute processing in univ-atts attribute group function.

AntennaHouse/pdf5-ml
https://github.com/AntennaHouse/pdf5-ml/blob/master/com.antennahouse.pdf5.ml/xsl/dita2fo_attribute.xsl 

       
       <xsl:copy-of select="$prmElement/@clear"/>

**Comment by raducoravu** [comment]
One possibility would be to override the common attributes processing in the "customFO.xsl", something like this:

      <xsl:template name="commonattributes">
        <xsl:apply-templates select="@id"/>
        <xsl:apply-templates select="*[contains(@class,' ditaot-d/ditaval-startprop ')] |
          *[contains(@class,' ditaot-d/ditaval-endprop ')]" mode="flag-attributes"/>
        <xsl:if test="contains(@outputclass, 'clear_')">
          <xsl:choose>
            <xsl:when test="contains(@outputclass, 'clear_left')">
              <xsl:attribute name="clear">left</xsl:attribute>
            </xsl:when>
            <xsl:when test="contains(@outputclass, 'clear_right')">
              <xsl:attribute name="clear">right</xsl:attribute>
            </xsl:when>
          </xsl:choose>
        </xsl:if>
      </xsl:template>

**Comment by raducoravu** [comment]
Possible values for the @clear attribute:
http://www.datypic.com/sc/fo11/a-clear-1.html

One more observation, Apache FOP does not support the clear attribute:

https://xmlgraphics.apache.org/fop/compliance.html

RenderX and Antenna House do.

**Comment by ToshihikoMakita** [comment]
> Possible values for the @clear attribute:
> http://www.datypic.com/sc/fo11/a-clear-1.html

Yes, it would be better to implement clear="both". I always use it to clear the side-floats.

**Comment by lionelmoi** [comment]
I too have encountered this issue. Do you plan to have a fix?

**Comment by raducoravu** [comment]
I do not have plans for a fix right now. If you want you can try one based on the observations I made on this issue.
